src/main.c

```diff
@@ -60,6 +61,45 @@ static void quick_shutdown(struct ev_loop *loop, ev_signal *w, int revents) {
     ev_break(loop, EVBREAK_ALL);
 }
 
+static void dump_shard_map(struct ev_loop *loop, ev_signal *w, int revents) {
+    stats_log("main: received SIGUSR1, dumping shard_map.");
+
+    FILE *f;
+    int fd;
+    // TODO: no hardcode path
+    if ((fd = open("/tmp/shard_map.json", O_RDWR|O_CREAT, 0644)) == -1) {
+        stats_error_log("dump_shard_map: failed to open or create /tmp/shard_map.json");
+        return;
+    }
+
+    if ((f = fdopen(fd, "r+")) == NULL) {
+        stats_error_log("dump_shard_map: fdopen failed");
+        close(fd);
+        return;
+    }
+
+    struct config *cfg = (struct config *) w->data;
+    list_t ring = cfg->statsd_config.ring;
+
+    json_t *out_object = json_object();
+    json_t *shard_map = json_array();
```

Comment:
This is never freed, memory leak?

Reply:
or does the json_delete on out_object recursively delete everything inside of it?